### PR TITLE
Feature/dependencies bundles

### PR DESCRIPTION
**Pull request by mmoreram**
ping @elcodi/owners 

This issue does not introduce any kind of BC Break and is related to https://github.com/symfony/symfony/pull/13945

### CODE REVIEW

File: src/Elcodi/Bundle/AttributeBundle/ElcodiAttributeBundle.php [hunk 1]
```diff
@@ -48,4 +49,16 @@ public function getContainerExtension()
...
+
+    /**
+     * Create instance of current bundle, and return dependent bundle namespaces
+     *
+     * @return array Bundle instances
+     */
+    public static function getBundleDependencies()
+    {
+        return [
+            '\Elcodi\Bundle\CoreBundle\ElcodiCoreBundle',
```
Comment (xphere) [inline src/Elcodi/Bundle/AttributeBundle/ElcodiAttributeBundle.php]: Can't wait to start supporting PHP >= 5.5 to make this like `ElcodiCoreBundle::class`

Comment (mmoreram) [inline src/Elcodi/Bundle/AttributeBundle/ElcodiAttributeBundle.php]: Absolutely :)

File: src/Elcodi/Bundle/CartBundle/Resources/config/eventListeners.yml [hunk 2]
```diff
@@ -11,7 +11,7 @@ services:
             - @elcodi.cart.manager
             - @elcodi.wrapper.currency
             - @elcodi.converter.currency
-            - @elcodi.manager.configuration
+            - "@=elcodi_config('product.use_stock', false)"
```
Comment (xphere) [inline src/Elcodi/Bundle/CartBundle/Resources/config/eventListeners.yml]: What's with the `false`? `elcodi_config` has no second parameter in expression language.

File: src/Elcodi/Bundle/CoreBundle/ElcodiCoreBundle.php [hunk 3]
```diff
@@ -40,4 +41,14 @@ public function getContainerExtension()
...
     }
+
+    /**
+     * Create instance of current bundle, and return dependent bundle namespaces
+     *
+     * @return array Bundle instances
+     */
+    public static function getBundleDependencies()
+    {
+        return [];
```
Comment (xphere) [inline src/Elcodi/Bundle/CoreBundle/ElcodiCoreBundle.php]: If this bundle doesn't depend on anything, is inheritance from `DependentBundleInterface` still relevant?

File: src/Elcodi/Bundle/MediaBundle/ElcodiMediaBundle.php [hunk 4]
```diff
@@ -48,4 +49,17 @@ public function getContainerExtension()
...
+
+    /**
+     * Create instance of current bundle, and return dependent bundle namespaces
+     *
+     * @return array Bundle instances
+     */
+    public static function getBundleDependencies()
+    {
+        return [
+            '\Elcodi\Bundle\CoreBundle\ElcodiCoreBundle',
```
Comment (xphere) [inline src/Elcodi/Bundle/MediaBundle/ElcodiMediaBundle.php]: If we want to be uniform, Elcodi bundles should go after third party ones.

File: src/Elcodi/Bundle/TestCommonBundle/Functional/Abstracts/AbstractElcodiKernel.php [hunk 5]
```diff
@@ -88,4 +64,8 @@ public function getLogDir()
         DIRECTORY_SEPARATOR .
         $this->getContainerClass() . '/Log/';
     }
+
```
Comment (xphere) [inline src/Elcodi/Bundle/TestCommonBundle/Functional/Abstracts/AbstractElcodiKernel.php]: Comment for nothing? :stuck_out_tongue:

File: src/Elcodi/Bundle/CartBundle/Resources/config/eventListeners.yml (hunk 2 above)
Comment (xphere) [inline src/Elcodi/Bundle/CartBundle/Resources/config/eventListeners.yml]: Ok, I stand corrected, this PR add also support for default values. :clap::clap::clap:

**Comment by xphere** [comment]
I like it! This is a clever move for Elcodi, and will be an easier transition when symfony accept bundle dependencies. Bravo! :clap: :dancer:

**Pull request merged**